synfig-core/src/synfig/palette.cpp

```diff
@@ -431,3 +431,17 @@ Palette::load_from_file(const synfig::filesystem::Path& filename)
 
 	return ret;
 }
+
+bool
+Palette::is_color_present(const Color& color)
+{
+        iterator iter;
+
+	for(iter=begin();iter!=end();++iter)
```

Comment:
The current approach requires traversing through the vector for every color addition.
Seems fine to me though, but we may maintain a hashmap if we need further optimization. Thoughts are welcome.

Reply:
I agree that traversing the vector for each color addition works fine for now. However, I think maintaining a hashmap could be a more efficient solution in the long term, especially if the number of colors grows significantly. It would allow for quicker lookups to check for duplicates.